controllers/selfnoderemediation_controller.go

```diff
@@ -293,18 +338,7 @@ func (r *SelfNodeRemediationReconciler) remediateWithResourceDeletion(snr *v1alp
 
 	r.logger.Info("done deleting node resources", "node name", node.Name)
 
-	fencingCompleted := fencingCompletedPhase
-	snr.Status.Phase = &fencingCompleted
-	if err := r.Client.Status().Update(context.Background(), snr); err != nil {
-		if apiErrors.IsConflict(err) {
-			// conflicts are expected since all self node remediation deamonset pods are competing on the same requests
-			return ctrl.Result{RequeueAfter: 1 * time.Second}, nil
-		}
-		r.logger.Error(err, "failed to mark SNR as fencing completed")
-		return ctrl.Result{}, err
-	}
-
-	return ctrl.Result{}, nil
+	return ctrl.Result{}, r.updateSnrRemediationPhase(fencingCompletedPhase, snr)
```

Comment:
At the moment we don't explicitly use rebootCompletedPhase (except from setting it).
IMO the code will be clearer by conditioning the resource deletion part to be at that stage.